linked_list.py

```diff
@@ -1,101 +1,107 @@
 #  _*_ coding utf-8 _*_
+"""Implement a Linked List."""
 
 
 class Node(object):
     """Node establishes functions that move within the list."""
 
-    def __init__(self, data=None, next_node=None):
+    def __init__(self, val=None, next_node=None):
         """Initialize the list."""
-        self.data = data
+        self.val = val
         self.next_node = next_node
 
-    def find_data(self):
-        """Return the data."""
-        return self.data
-
-    def find_next_node(self):
-        """Return the next node."""
-        return self.next_node
-
-    def set_next_node(self, new_next):
-        """Reset next node."""
-        self.next_node = new_next
-
 
 class LinkedList(object):
-    """Manipulate the linked list."""
+    """Linked List Implementation."""
 
-    def __init__(self):
+    def __init__(self, lst=[]):
         """Initialize the head of the node to (None is not given)."""
         self.head = None
-        self.size = 0
+        self.length = 0
+        for val in lst:
+            self.insert(val)
 
-    def insert_node(self, data):
-        """Insert new node."""
-        new_node = Node(data)
+    def insert(self, val):
+        """Insert new node to Linked List."""
+        new_node = Node(val)
         if self.head is None:
             self.head = new_node
         else:
             new_node.next_node = self.head
             self.head = new_node
-        self.size += 1
-        # print(new_node.data)
-        return new_node.data
-
-    def get_size(self):
-        """Get the total count of nodes in list."""
-        return self.size
+        self.length += 1
+        return new_node.val
 
-    def search(self, data):
-        """Search for a specific node in list."""
-        current_node = self.head
-        found = False
-        while current_node and found is False:
-            if current_node.find_data() == data:
-                found = True
-            else:
-                current_node = current_node.find_next_node()
-        if current_node is None:
-            raise ValueError("Data not in List")
-        return current_node
+    def pop(self):
+        """Pop the head and return its value."""
+        old_head = self.head
+        self.head = self.head.next_node
+        self.length -= 1
+        return old_head.val
 
-    def remove_node(self, data):
-        """Remove a specific node in list."""
+    def size(self):
+        """Get the total count of nodes in list."""
+        return self.length
+
+    def search(self, val):
+        """Search for a specific node in list, if not present return None."""
+        try:
+            current_node = self.head
+            while True:
```

Comment:
If you change this to `while current_node:`, you can avoid the entire try...except block.